behaviour/graql/reasoner/resolution/concept-inequality.feature

```diff
@@ -0,0 +1,402 @@
+#
+
+# TODO: re-enable all steps in file when 3-hop transitivity is resolvable
+Feature: Concept Inequality Resolution
+
+  Background: Set up keyspaces for resolution testing
+
+    Given connection has been opened
+    Given connection delete all keyspaces
+    Given connection open sessions for keyspaces:
+      | materialised |
+      | reasoned     |
+    Given materialised keyspace is named: materialised
+    Given reasoned keyspace is named: reasoned
+    Given for each session, graql define
+      """
+      define
+
+      ball sub entity,
+          has name,
+          plays ball1,
+          plays ball2;
+
+      # Represents a selection of balls from a bag, with replacement after each selection
+      selection sub relation,
+          relates ball1,
+          relates ball2;
+
+      name sub attribute, value string;
+
+      transitivity sub rule,
+      when {
+          (ball1:$x, ball2:$y) isa selection;
+          (ball1:$y, ball2:$z) isa selection;
+      },
+      then {
+          (ball1:$x, ball2:$z) isa selection;
+      };
+      """
+    Given for each session, graql insert
+      """
+      insert
+
+      $a isa ball, has name 'a';
+      $b isa ball, has name 'b';
+      $c isa ball, has name 'c';
+
+      # selection is effectively reflexive, symmetric and transitive
+      (ball1: $a, ball2: $b) isa selection;
+      (ball1: $b, ball2: $a) isa selection;
+      (ball1: $b, ball2: $c) isa selection;
+      (ball1: $c, ball2: $b) isa selection;
+      """
+
+
+  # TODO: re-enable all steps when 3-hop transitivity is resolvable
+  Scenario: a rule can be applied based on concept inequality
+    Given for each session, graql define
+      """
+      define
+
+      state sub entity,
+          plays related-state,
+          has name;
+
+      achieved sub relation,
+          relates related-state;
+
+      prior sub relation,
+          relates related-state;
+
+      holds sub relation,
+          relates related-state;
+
+      state-rule sub rule,
+      when {
+          $st isa state;
+          (related-state: $st) isa achieved;
+          (related-state: $st2) isa prior;
+          $st != $st2;
+      },
+      then {
+          (related-state: $st) isa holds;
+      };
+      """
+    Given for each session, graql insert
+      """
+      insert
+
+      $s1 isa state, has name 's1';
+      $s2 isa state, has name 's2';
+
+      (related-state: $s1) isa prior;
+      (related-state: $s1) isa achieved;
+      (related-state: $s2) isa achieved;
+      """
+#    When materialised keyspace is completed
+    Then for graql query
+      """
+      match (related-state: $s) isa holds; get;
+      """
+#    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 1
+    Then for graql query
+      """
+      match
+        $s isa state, has name 's2';
+        (related-state: $s) isa holds;
+      get;
+      """
+#    Then all answers are correct in reasoned keyspace
```

Comment:
we can make the intent of the tests clearer by adding a step to assert that the answer sets are the same, rather than using a Graql conjunction to do that for us

Reply:
Added 'answer set is equivalent for graql query` BDD step